be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -188,14 +188,32 @@ Status HdfsTextScanner::parse_csv(int chunk_size, ChunkPtr* chunk) {
 
         bool has_error = false;
         int num_materialize_columns = _scanner_params.materialize_slots.size();
+        int field_size = fields.size();
+        if (_scanner_params.hive_column_names->size() != field_size) {
+            LOG(WARNING) << strings::Substitute("Size mismatch between hive column $0 names and fields $1!",
+                                                _scanner_params.hive_column_names->size(), fields.size());
+        }
         for (int j = 0; j < num_materialize_columns; j++) {
             int index = _scanner_params.materialize_index_in_chunk[j];
-            const Slice& field = fields[_scanner_params.materialize_slots[j]->id() - 1];
-            options.type_desc = &(_scanner_params.materialize_slots[j]->type());
-            if (!_converters[j]->read_string(_column_raw_ptrs[index], field, options)) {
-                chunk->get()->set_num_rows(num_rows);
-                has_error = true;
-                break;
+            int column_field_index = _columns_index[_scanner_params.materialize_slots[j]->col_name()];
+            Column* column = _column_raw_ptrs[index];
```

Comment:
Is size of `_columns_index` always equal to size of `_column_raw_ptrs`? should we check if `index` exceed upper bound of it?